pytests/src/exceptions.rs

```diff
@@ -0,0 +1,72 @@
+use pyo3::exceptions::PyException;
+use pyo3::prelude::*;
+use pyo3::types::{PyDict, PyTuple};
+
+#[pyclass(extends=PyException, subclass)]
+struct CustomException {}
+
+#[pymethods]
+impl CustomException {
+    #[new(signature = (*_args, **_kwargs))]
```

Comment:
I wrote a bug here which confused me for a bit; `#[new]` ignores `signature`.

Either this should be a hard error or it should be equivalent to `#[new] #[pyo3(signature = (*_args, **_kwargs))]`.